Source/WebCore/dom/FullscreenManager.cpp

```diff
@@ -713,11 +718,26 @@ void FullscreenManager::setAnimatingFullscreen(bool flag)
     m_isAnimatingFullscreen = flag;
 }
 
+void FullscreenManager::updatePageFullscreenStatusIfTopDocument()
+{
+    RefPtr frame = m_document->frame();
+    if (!frame || !frame->isMainFrame())
+        return;
+
+    RefPtr protectedPage = frame->protectedPage();
+    if (!protectedPage)
+        return;
+
+    protectedPage->setTopDocumentHasFullscreenElement(m_fullscreenElement);
```

Comment:
This is a behavior change from before. Potentially an escape.

`m_fullscreenElement` does not track the case where the top document is implicitly fullscreened by its child frames but only when top document is explicitly fullscreened.

`fullscreenElement()` would have been the right thing to use. 

Also, this changeset doesn't appear to broadcast the fullscreen status in the implicit case.